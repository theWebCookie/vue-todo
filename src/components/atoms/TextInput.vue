<template>
  <input type="text" :placeholder="placeholder" :class="$style.searchBar" v-model="inputValue" @keyup.enter="addTask" />
</template>

<script>
export default {
  name: 'TextInput',
  props: {
    placeholder: {
      type: String,
      default: 'Add your task',
    },
  },
  data() {
    return {
      inputValue: '',
    };
  },
  methods: {
    addTask() {
      if (this.inputValue.trim()) {
        const task = {
          id: Date.now(),
          title: this.inputValue,
          completed: false,
        };
        this.$store.dispatch('addTask', task);
        this.inputValue = '';
      }
    },
  },
};
</script>

<style module lang="scss">
.searchBar {
  width: 100%;
  padding: $spacing-small 0;
  font-size: 16px;
  border: none;
}
</style>
