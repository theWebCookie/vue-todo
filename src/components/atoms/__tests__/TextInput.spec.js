import { shallowMount } from '@vue/test-utils';
import { createStore } from 'vuex';
import { describe, it, expect, jest, beforeEach } from '@jest/globals';
import TextInput from '../TextInput.vue';

describe('TextInput.vue', () => {
  let store;
  let actions;

  beforeEach(() => {
    actions = {
      addTask: jest.fn(),
    };

    store = createStore({
      actions,
    });
  });

  it('renders correctly and matches the snapshot', () => {
    const wrapper = shallowMount(TextInput, {
      global: {
        plugins: [store],
        mocks: {
          $style: {
            searchBar: 'searchBar',
          },
        },
      },
      props: {
        placeholder: 'Add new task',
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders with default placeholder', () => {
    const wrapper = shallowMount(TextInput, {
      global: {
        plugins: [store],
        mocks: {
          $style: {
            searchBar: 'searchBar',
          },
        },
      },
    });

    const input = wrapper.find('input');
    expect(input.attributes('placeholder')).toBe('Add your task');
  });

  it('emits an action to add a task when Enter is pressed', async () => {
    const wrapper = shallowMount(TextInput, {
      global: {
        plugins: [store],
        mocks: {
          $style: {
            searchBar: 'searchBar',
          },
        },
      },
    });

    await wrapper.setData({ inputValue: 'New Task' });
    await wrapper.find('input').trigger('keyup.enter');

    expect(actions.addTask).toHaveBeenCalledWith(expect.any(Object), {
      id: expect.any(Number),
      title: 'New Task',
      completed: false,
    });

    expect(wrapper.vm.inputValue).toBe('');
  });
});
