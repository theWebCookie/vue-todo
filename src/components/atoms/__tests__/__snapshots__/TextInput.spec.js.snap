// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput.vue renders correctly and matches the snapshot 1`] = `
<input
  class="searchBar"
  placeholder="Add new task"
  type="text"
/>
`;
